utils/biospecimen.js

```diff
@@ -212,12 +212,11 @@ const biospecimenAPIs = async (req, res) => {
             return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
         }
         if(req.query.masterSpecimenId) {
-            const masterSpecimenId = req.query.masterSpecimenId;
-            if(!masterSpecimenId) return res.status(400).json(getResponseJSON('Bad request!', 400));
             const { searchSpecimen } = require('./firestore');
-            const allSitesFlag = (req.query.allSitesFlag) ? true : false
+            const masterSpecimenId = req.query.masterSpecimenId;
+            const allSitesFlag = (req.query.allSitesFlag) ? true : false;
             const response = await searchSpecimen(masterSpecimenId, siteCode, allSitesFlag);
-            if(!response) return res.status(404).json(getResponseJSON('Data not found!', 404));
+            if (!response) return res.status(200).json({data: {}, code:200});
```

Comment:
might suggest combining the `200` return statements into one line that is always going to return something for data and code, then let `searchSpecimen` handle either returning a populated object or an empty object. Also would suggest having error handling in `searchSpecimen` and then checking here `if(response instanceof Error)`

Reply:
I have a question if `searchSpecimen` function has an internal error what will be the result of `response`?

Reply:
Combined returns, and added error handler.